### PR TITLE
remove machineRef field from host crd

### PR DESCRIPTION
**Pull request by dhellmann**

### CODE REVIEW

File: pkg/apis/metal3/v1alpha1/baremetalhost_types_test.go [hunk 1]
```diff
@@ -37,22 +37,6 @@ func TestHostAvailable(t *testing.T) {
 			Expected:    false,
 			FailMessage: "host with error returned as available",
 		},
-		{
```
Comment (mhrivnak) [inline pkg/apis/metal3/v1alpha1/baremetalhost_types_test.go]: Why remove this test case?

File: pkg/apis/metal3/v1alpha1/baremetalhost_types_test.go [hunk 2]
```diff
@@ -113,19 +97,6 @@ func TestHostNeedsHardwareInspection(t *testing.T) {
 		t.Error("expected to not need hardware inspection")
 	}
 
-	hostWithMachine := BareMetalHost{
```
Comment (mhrivnak) [inline pkg/apis/metal3/v1alpha1/baremetalhost_types_test.go]: Why remove this test case?

File: pkg/apis/metal3/v1alpha1/baremetalhost_types_test.go (hunk 1 above)
Comment (dhellmann) [reply pkg/apis/metal3/v1alpha1/baremetalhost_types_test.go]: A previous PR that added the consumerRef field added a redundant test in parallel. This one references the field that was removed, so the test can be removed without losing coverage.

File: pkg/apis/metal3/v1alpha1/baremetalhost_types_test.go (hunk 2 above)
Comment (dhellmann) [reply pkg/apis/metal3/v1alpha1/baremetalhost_types_test.go]: Ditto.

**Pull request merged**